weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/tsDataModelHooksEvaluationComparison.ts

```diff
@@ -548,6 +482,532 @@ const fetchEvaluationComparisonResults = async (
+  evalCall: any,
+  evalCallId: string,
+  output: any
+): void => {
+  // In imperative evaluations, the metrics can appear directly in the output object
+  // We need to recursively process them to extract all metrics
+  const recursiveAddMetric = (metricVal: any, currPath: string[]) => {
+    if (typeof metricVal === 'boolean') {
+      const metricDimension: MetricDefinition = {
+        scoreType: 'binary',
+        metricSubPath: currPath,
+        source: 'derived', // Using 'derived' since we don't have a direct scorer reference
+      };
+      const metricId = metricDefinitionId(metricDimension);
+      result.summaryMetrics[metricId] = metricDimension;
+      evalCall.summaryMetrics[metricId] = {
+        value: metricVal,
+        sourceCallId: evalCallId,
+      };
+    } else if (typeof metricVal === 'number') {
+      const metricDimension: MetricDefinition = {
+        scoreType: 'continuous',
+        metricSubPath: currPath,
+        source: 'derived',
+      };
+      const metricId = metricDefinitionId(metricDimension);
+      result.summaryMetrics[metricId] = metricDimension;
+      evalCall.summaryMetrics[metricId] = {
+        value: metricVal,
+        sourceCallId: evalCallId,
+      };
+    } else if (
+      metricVal != null &&
+      typeof metricVal === 'object' &&
+      !Array.isArray(metricVal)
+    ) {
+      // Special case for auto-summarized binary scores
+      if (isBinarySummaryScore(metricVal)) {
+        const metricDimension: MetricDefinition = {
+          scoreType: 'binary',
+          metricSubPath: currPath,
+          source: 'derived',
+        };
+        const metricId = metricDefinitionId(metricDimension);
+        result.summaryMetrics[metricId] = metricDimension;
+        evalCall.summaryMetrics[metricId] = {
+          value: metricVal.true_fraction,
+          sourceCallId: evalCallId,
+        };
+      }
+      // Special case for auto-summarized continuous scores
+      else if (isContinuousSummaryScore(metricVal)) {
+        const metricDimension: MetricDefinition = {
+          scoreType: 'continuous',
+          metricSubPath: currPath,
+          source: 'derived',
+        };
+        const metricId = metricDefinitionId(metricDimension);
+        result.summaryMetrics[metricId] = metricDimension;
+        evalCall.summaryMetrics[metricId] = {
+          value: metricVal.mean,
+          sourceCallId: evalCallId,
+        };
+      }
+      // Otherwise, process nested objects
+      else {
+        Object.entries(metricVal).forEach(([key, val]) => {
+          recursiveAddMetric(val, [...currPath, key]);
+        });
+      }
+    }
+  };
+
+  // Start processing from the root of the output
+  Object.entries(output).forEach(([key, val]) => {
+    recursiveAddMetric(val, [key]);
+  });
+};
+
+const isImperativeEvalCall = (call: TraceCallSchema) => {
+  return call.attributes?._weave_eval_meta?.imperative;
+};
+
+const generateStableDigest = (obj: any): string => {
```

Comment:
in a super ideal world, you use the same digest calculation that the backend uses so that you can compare across evaluation types (but could be a TODO - might want to mark it)

Reply:
TODO: Find the reference in the server and share